### ISSUE TITLE
Provide bindings for popular frontend frameworks as subprojects

### ISSUE THREAD

**Issue by kripod** [material-components/material-components-web#145]
While framework examples are not hard to follow or even copy, they introduce extra boilerplate code for developers. I would suggest creating separate packages for each popular frontend framework (e.g. `react-material-components-web`), potentially raising the adoption rate of the library.

**Comment by sgomes** [comment]
Thanks, @kripod! This is actually something we hope to inspire the community to do, by providing our barebones examples. 🙂 

We would definitely like to see separate packages for each framework, and that's one of the main reasons why we spent so much time working on our architecture and component modularity. However, we're not experts on these frameworks, as you can probably tell from the PRs we've already received to our examples in the few weeks we've been live. 😉 In addition, we're a stretched a bit thin as it it, as you can probably tell from the huge list of TBD components in our issue tracker. 😅 

We think each framework's community is much more well-equipped to develop and maintain their own set of bindings, since they know their framework inside-out and can easily follow best practices and update things as new releases come out. The awesome @petejohanson has already [started that work for Vue](https://github.com/petejohanson/vue-mdc), for example!

As MDC-Web and each set of bindings matures, I can definitely see us linking out to and recommending these projects, as well as perhaps getting rid of the corresponding framework examples.

In the meantime, if anyone starts a set of bindings for any framework, please let us know! We'd love to keep an eye on your project and help out wherever we can. We also want to make sure that our architecture is solid enough to support all frameworks and make changes to it if not, and the earlier we can do that, the better!

**Comment by kripod** [comment]
It seems like a [React implementation](https://github.com/haoxins/react-mdc) is on the way, too... 😊

**Comment by Thanood** [comment]
I did the example for the Aurelia framework and also started work on a more complete example (possible to evolve into a "real" set of components). Unfortunately, I have some other projects that need attention and so my time is limited at the moment.

If anyone would like to take that on I'd be glad to collaborate. 😃 

MDC did grab some attention in the Aurelia community (or at least that's my perception) so I think there will be a project relatively soon.
I guess Angular, with its much bigger community, should see a library even sooner.

Personally I think that MDC is a great project and I think it deserves attention. And a good set of framework wrappers. 😉

**Comment by traviskaufman** [comment]
Closing this issue, since as @sgomes mentioned we'd rather have framework bindings be written by authors whose area of expertise is that framework. To that end, _we are here to help for anyone that would like to write a binding_. If you're interested in doing so, please let us know either on this issue or by opening a new issue detailing any problems you're running into. :)

**Comment by kripod** [comment]
I just started working on [material-components-react](https://github.com/kripod/material-components-react), all contributions are appreciated!
